// @inrupt/lit-generated-vocab-common/index.ts
export { SCHEMA_INRUPT_EXT } from "./GeneratedVocab/SCHEMA_INRUPT_EXT";
export { VCARD } from "./GeneratedVocab/VCARD";
export { VCARD_INRUPT_EXT } from "./GeneratedVocab/VCARD_INRUPT_EXT";
export { RDF } from "./GeneratedVocab/RDF";
export { RDF_INRUPT_EXT } from "./GeneratedVocab/RDF_INRUPT_EXT";
export { RDFS } from "./GeneratedVocab/RDFS";
export { XSD } from "./GeneratedVocab/XSD";
export { OWL } from "./GeneratedVocab/OWL";
export { LDP } from "./GeneratedVocab/LDP";
export { HTTP } from "./GeneratedVocab/HTTP";
export { HTTPH } from "./GeneratedVocab/HTTPH";
export { OLO } from "./GeneratedVocab/OLO";
export { SKOS } from "./GeneratedVocab/SKOS";
export { UI } from "./GeneratedVocab/UI";
export { VANN } from "./GeneratedVocab/VANN";
export { DCTERMS } from "./GeneratedVocab/DCTERMS";
export { AS } from "./GeneratedVocab/AS";
export { RLOG } from "./GeneratedVocab/RLOG";
export { FOAF } from "./GeneratedVocab/FOAF";
export { QB } from "./GeneratedVocab/QB";
export { SDMX_DIMENSION } from "./GeneratedVocab/SDMX_DIMENSION";
