// @inrupt/lit-generated-vocab-common/GeneratedVocab/RDF_INRUPT_EXT.ts
import { LitVocabTerm, getLocalStore } from "@inrupt/lit-vocab-term";
import dataFactory from "@rdfjs/data-model";

const { namedNode } = dataFactory;
import { NamedNode } from "rdf-js";

/**
  Generated by artifact generator [@inrupt/lit-artifact-generator], version [0.10.15]
  as part of artifact: [vocab-common], version: [0.3.11]
  at 'Saturday, May 30, 2020 1:18 AM'.

  Vocabulary built from vocab list file: [/home/pmcb55/Work/Projects/LIT/lit-vocab/vocab/lit-rdf-vocab/Common/Vocab-List-LIT-Common.yml].
 */

function _NS(localName: string): NamedNode { 
  return namedNode(`http://www.w3.org/1999/02/22-rdf-syntax-ns#${localName}`);
}

const RDF_INRUPT_EXT = {
  PREFIX: "rdf-inrupt-ext",
  NAMESPACE: "http://www.w3.org/1999/02/22-rdf-syntax-ns#",
  PREFIX_AND_NAMESPACE: { "rdf-inrupt-ext": "http://www.w3.org/1999/02/22-rdf-syntax-ns#" },
  NS: _NS,


  // *******************
  // All the Properties.
  // *******************

  /**
   * The subject is an instance of a class.
   */
  type: new LitVocabTerm(
    _NS("type"),
    dataFactory,
    getLocalStore(),
    false
  )
    .addLabel(`Tipo`, "es")
    .addLabelNoLanguage(`type`)
    .addCommentNoLanguage(`The subject is an instance of a class.`),

};

export { RDF_INRUPT_EXT };
