// @inrupt/lit-generated-vocab-common/GeneratedVocab/SCHEMA_INRUPT_EXT.ts
import { LitVocabTerm, getLocalStore } from "@inrupt/lit-vocab-term";
import dataFactory from "@rdfjs/data-model";

const { namedNode } = dataFactory;
import { NamedNode } from "rdf-js";

/**
  Generated by artifact generator [@inrupt/lit-artifact-generator], version [0.10.15]
  as part of artifact: [vocab-common], version: [0.3.11]
  at 'Saturday, May 30, 2020 1:18 AM'.

  Vocabulary built from vocab list file: [/home/pmcb55/Work/Projects/LIT/lit-vocab/vocab/lit-rdf-vocab/Common/Vocab-List-LIT-Common.yml].

  Extension to Schema.org terms providing multilingual alternative names and translations for comments (e.g. for use directly as labels or tool-tips in user interfaces or error messages)
 */

function _NS(localName: string): NamedNode { 
  return namedNode(`http://schema.org/${localName}`);
}

const SCHEMA_INRUPT_EXT = {
  PREFIX: "schema-inrupt-ext",
  NAMESPACE: "http://schema.org/",
  PREFIX_AND_NAMESPACE: { "schema-inrupt-ext": "http://schema.org/" },
  NS: _NS,

  // *****************
  // All the Classes.
  // *****************

  /**
   * A person (alive, dead, undead, or fictional).
   */
  Person: new LitVocabTerm(
    _NS("Person"),
    dataFactory,
    getLocalStore(),
    false
  )
    .addLabel(`Person`, "en")
    .addLabel(`La personne`, "fr")
    .addLabel(`Person`, "de")
    .addLabel(`Persona`, "es")
    .addLabel(`Persona`, "it")
    .addLabelNoLanguage(`Person`)
    .addComment(`Une personne (vivante, morte, mort-vivant ou fictive).`, "fr")
    .addComment(`Eine Person (lebendig, tot, untot oder fiktiv).`, "de")
    .addComment(`Una persona (viva, muerta, no muerta o ficticia).`, "es")
    .addComment(`Una persona (viva, morta, non morta o immaginaria).`, "it")
    .addCommentNoLanguage(`A person (alive, dead, undead, or fictional).`),

  // *******************
  // All the Properties.
  // *******************

  /**
   * An alias for the item.
   */
  alternateName: new LitVocabTerm(
    _NS("alternateName"),
    dataFactory,
    getLocalStore(),
    false
  )
    .addLabel(`Alternate Name`, "en")
    .addLabel(`Nom alternatif`, "fr")
    .addLabel(`Anderer Name`, "de")
    .addLabel(`Nombre alternativo`, "es")
    .addLabel(`Nome alternativo`, "it")
    .addLabelNoLanguage(`alternateName`)
    .addComment(`Un alias pour l'article.`, "fr")
    .addComment(`Ein Alias ​​für den Artikel.`, "de")
    .addComment(`Un alias para el elemento.`, "es")
    .addComment(`Un alias per l'elemento.`, "it")
    .addCommentNoLanguage(`An alias for the item.`),

  /**
   * Given name. In the U.S., the first name of a Person. This can be used along with familyName instead of the name property.
   */
  givenName: new LitVocabTerm(
    _NS("givenName"),
    dataFactory,
    getLocalStore(),
    false
  )
    .addLabel(`Given Name`, "en")
    .addLabel(`Prénom`, "fr")
    .addLabel(`Vorname`, "de")
    .addLabel(`Nombre de pila`, "es")
    .addLabel(`Nome di battesimo`, "it")
    .addLabelNoLanguage(`givenName`)
    .addComment(`Prénom. Aux États-Unis, le prénom d’une personne. Ceci peut être utilisé avec familyName au lieu de la propriété name.`, "fr")
    .addComment(`Vorname. In den USA der Vorname einer Person. Dies kann zusammen mit familyName anstelle der Eigenschaft name verwendet werden.`, "de")
    .addComment(`Nombre de pila. En los EE. UU., El primer nombre de una persona. Esto se puede usar junto con familyName en lugar de la propiedad name.`, "es")
    .addComment(`Nome di battesimo. Negli Stati Uniti, il primo nome di una persona. Questo può essere usato insieme a familyName al posto della proprietà name.`, "it")
    .addCommentNoLanguage(`Given name. In the U.S., the first name of a Person. This can be used along with familyName instead of the name property.`),

  /**
   * Family name. In the U.S., the last name of an Person. This can be used along with givenName instead of the name property.
   */
  familyName: new LitVocabTerm(
    _NS("familyName"),
    dataFactory,
    getLocalStore(),
    false
  )
    .addLabel(`Family Name`, "en")
    .addLabel(`Nom de famille`, "fr")
    .addLabel(`Nachname`, "de")
    .addLabel(`Apellido`, "es")
    .addLabel(`Cognome`, "it")
    .addLabelNoLanguage(`familyName`)
    .addComment(`Nom de famille. Aux États-Unis, le nom de famille d’une personne. Ceci peut être utilisé avec GivenName au lieu de la propriété name.`, "fr")
    .addComment(`Nachname. In den USA der Nachname einer Person. Dies kann zusammen mit givenName anstelle der Eigenschaft name verwendet werden.`, "de")
    .addComment(`Apellido. En los EE.UU., el apellido de una persona. Esto se puede usar junto con givenName en lugar de la propiedad name.`, "es")
    .addComment(`Cognome. Negli Stati Uniti, il cognome di una persona. Questo può essere usato insieme a givenName al posto della proprietà name.`, "it")
    .addCommentNoLanguage(`Family name. In the U.S., the last name of an Person. This can be used along with givenName instead of the name property.`),

  /**
   * An additional name for a Person, can be used for a middle name.
   */
  additionalName: new LitVocabTerm(
    _NS("additionalName"),
    dataFactory,
    getLocalStore(),
    false
  )
    .addLabel(`Additional Name`, "en")
    .addLabel(`Nom additionnel`, "fr")
    .addLabel(`Zusätzlicher Name`, "de")
    .addLabel(`Nombre adicional`, "es")
    .addLabel(`Nome aggiuntivo`, "it")
    .addLabelNoLanguage(`additionalName`)
    .addComment(`Un nom supplémentaire pour une personne peut être utilisé pour un deuxième prénom.`, "fr")
    .addComment(`Ein zusätzlicher Name für eine Person kann für einen zweiten Vornamen verwendet werden.`, "de")
    .addComment(`Un nombre adicional para una persona, se puede utilizar para un segundo nombre.`, "es")
    .addComment(`Un nome aggiuntivo per una persona può essere usato per un secondo nome.`, "it")
    .addCommentNoLanguage(`An additional name for a Person, can be used for a middle name.`),

  /**
   * A license document that applies to this content, typically indicated by URL.
   */
  license: new LitVocabTerm(
    _NS("license"),
    dataFactory,
    getLocalStore(),
    false
  )
    .addLabel(`License`, "en")
    .addLabel(`License`, "fr")
    .addLabel(`Lizenz`, "de")
    .addLabel(`Licencia`, "es")
    .addLabel(`Licenza`, "it")
    .addLabelNoLanguage(`license`)
    .addComment(`Un document de licence qui s'applique à ce contenu, généralement indiqué par une URL.`, "fr")
    .addComment(`Ein Lizenzdokument, das für diesen Inhalt gilt und in der Regel durch eine URL angegeben wird.`, "de")
    .addComment(`Un documento de licencia que se aplica a este contenido, normalmente indicado por URL.`, "es")
    .addComment(`Un documento di licenza che si applica a questo contenuto, generalmente indicato dall'URL.`, "it")
    .addCommentNoLanguage(`A license document that applies to this content, typically indicated by URL.`),

  /**
   * Must have comment too!
   */
  newTerm: new LitVocabTerm(
    _NS("newTerm"),
    dataFactory,
    getLocalStore(),
    false
  )
    .addLabelNoLanguage(`newTerm`)
    .addComment(`Must have comment too!`, "en"),

  /**
   * The name of the item.
   */
  name: new LitVocabTerm(
    _NS("name"),
    dataFactory,
    getLocalStore(),
    false
  )
    .addLabel(`Name`, "en")
    .addLabel(`Nom`, "fr")
    .addLabel(`Name`, "de")
    .addLabel(`Nombre`, "es")
    .addLabel(`Nome`, "it")
    .addLabelNoLanguage(`name`)
    .addComment(`Le nom de l'objet.`, "fr")
    .addComment(`Der Name des Artikels.`, "de")
    .addComment(`El nombre del artículo.`, "es")
    .addComment(`Il nome dell'articolo.`, "it")
    .addCommentNoLanguage(`The name of the item.`),

};

export { SCHEMA_INRUPT_EXT };
