// @inrupt/lit-generated-vocab-common/GeneratedVocab/VCARD_INRUPT_EXT.ts
import { LitVocabTerm, getLocalStore } from "@inrupt/lit-vocab-term";
import dataFactory from "@rdfjs/data-model";

const { namedNode } = dataFactory;
import { NamedNode } from "rdf-js";

/**
  Generated by artifact generator [@inrupt/lit-artifact-generator], version [0.10.15]
  as part of artifact: [vocab-common], version: [0.3.11]
  at 'Saturday, May 30, 2020 1:18 AM'.

  Vocabulary built from vocab list file: [/home/pmcb55/Work/Projects/LIT/lit-vocab/vocab/lit-rdf-vocab/Common/Vocab-List-LIT-Common.yml].
 */

function _NS(localName: string): NamedNode { 
  return namedNode(`http://www.w3.org/2006/vcard/ns#${localName}`);
}

const VCARD_INRUPT_EXT = {
  PREFIX: "vcard-inrupt-ext",
  NAMESPACE: "http://www.w3.org/2006/vcard/ns#",
  PREFIX_AND_NAMESPACE: { "vcard-inrupt-ext": "http://www.w3.org/2006/vcard/ns#" },
  NS: _NS,


  // *******************
  // All the Properties.
  // *******************

  /**
   * To specify the components of the delivery address for the object
   */
  hasAddress: new LitVocabTerm(
    _NS("hasAddress"),
    dataFactory,
    getLocalStore(),
    false
  )
    .addLabel(`Address`, "en")
    .addLabel(`Dirección`, "es")
    .addComment(`To specify the components of the delivery address for the object`, "en"),

  /**
   * To specify the electronic mail address for communication with the object
   */
  hasEmail: new LitVocabTerm(
    _NS("hasEmail"),
    dataFactory,
    getLocalStore(),
    false
  )
    .addLabel(`EMail`, "en")
    .addLabel(`Correo electrónico`, "es")
    .addComment(`To specify the electronic mail address for communication with the object`, "en"),

  /**
   * To specify the telephone number for telephony communication with the object
   */
  hasTelephone: new LitVocabTerm(
    _NS("hasTelephone"),
    dataFactory,
    getLocalStore(),
    false
  )
    .addLabel(`Telephone`, "en")
    .addLabel(`Teléfono`, "es")
    .addComment(`To specify the telephone number for telephony communication with the object`, "en"),

  /**
   * The formatted text corresponding to the name of the object
   */
  fn: new LitVocabTerm(
    _NS("fn"),
    dataFactory,
    getLocalStore(),
    false
  )
    .addLabel(`Name`, "en")
    .addLabel(`Nombre`, "es")
    .addComment(`The formatted text corresponding to the name of the object`, "en"),

  /**
   * To specify the function or part played in a particular situation by the object
   */
  role: new LitVocabTerm(
    _NS("role"),
    dataFactory,
    getLocalStore(),
    false
  )
    .addLabel(`Role`, "en")
    .addLabel(`Rol`, "es")
    .addComment(`To specify the function or part played in a particular situation by the object`, "en"),

  /**
   * To specify the organizational name associated with the object
   */
  organization_name: new LitVocabTerm(
    _NS("organization-name"),
    dataFactory,
    getLocalStore(),
    false
  )
    .addLabel(`Company`, "en")
    .addLabel(`Compañía`, "es")
    .addComment(`To specify the organizational name associated with the object`, "en"),

  /**
   * The street address associated with the address of the object
   */
  street_address: new LitVocabTerm(
    _NS("street-address"),
    dataFactory,
    getLocalStore(),
    false
  )
    .addLabel(`Dirección`, "es")
    .addLabel(`Street Address`, "en")
    .addComment(`The street address associated with the address of the object`, "en"),

  /**
   * The locality (e.g. city or town) associated with the address of the object
   */
  locality: new LitVocabTerm(
    _NS("locality"),
    dataFactory,
    getLocalStore(),
    false
  )
    .addLabel(`Locality`, "en")
    .addLabel(`Localidad`, "es")
    .addComment(`The locality (e.g. city or town) associated with the address of the object`, "en"),

  /**
   * The postal code associated with the address of the object
   */
  postal_code: new LitVocabTerm(
    _NS("postal-code"),
    dataFactory,
    getLocalStore(),
    false
  )
    .addLabel(`Postal Code`, "en")
    .addLabel(`Código Postal`, "es")
    .addComment(`The postal code associated with the address of the object`, "en"),

  /**
   * The region (e.g. state or province) associated with the address of the object
   */
  region: new LitVocabTerm(
    _NS("region"),
    dataFactory,
    getLocalStore(),
    false
  )
    .addLabel(`Region`, "en")
    .addLabel(`Región`, "es")
    .addComment(`The region (e.g. state or province) associated with the address of the object`, "en"),

  /**
   * The country name associated with the address of the object
   */
  country_name: new LitVocabTerm(
    _NS("country-name"),
    dataFactory,
    getLocalStore(),
    false
  )
    .addLabel(`Country`, "en")
    .addLabel(`Pais`, "es")
    .addComment(`The country name associated with the address of the object`, "en"),

};

export { VCARD_INRUPT_EXT };
